internal/internal_event_handlers.go

```diff
@@ -1457,6 +1462,11 @@ func (weh *workflowExecutionEventHandlerImpl) handleWorkflowExecutionStarted(
 	// as the value varies by WFT)
 	weh.sdkFlags.tryUse(SDKFlagProtocolMessageCommand, !weh.isReplay)
 
+	// Use the first execution run ID from the start event as the initial seed.
+	// First execution run ID stays the same for the entire chain of workflow resets.
+	// This helps us keep child workflow IDs consistent up until the next reset point.
+	weh.workflowInfo.childWorkflowIDSeed = attributes.GetFirstExecutionRunId()
```

Comment:
Can FirstExecutionRunID be empty ever? The current version of the server is guaranteed to include it in the start event. But could there be some long running WFs with this field being empty in start event?
Asking to see if we want to be defensive here and fall back to RunID if FirstExecutionRunID is empty.

Reply:
We need to set `childWorkflowIDSeed` on `WorkflowInfo` in the same place where we set the `WorkflowExecution` (`createWorkflowContext` which has the started event).

I think whether `attributes.FirstExecutionRunID` is always present is a server-side question. But I do not think it is accurate to use here because it is carried over in other situations like cron and workflow retry. You may need to confirm with Java and/or Core on what they use as the first run ID here. It may need to be `OriginalExecutionRunId` and you may need to fall back to the task's workflow execution run ID if that's not present.

Reply:
Java uses `original_execution_run_id`
```
    // This is the run id when the WorkflowExecutionStarted event was written.
    // A workflow reset changes the execution run_id, but preserves this field.
    string original_execution_run_id = 14;
```

Reply:
ok. Using `original_execution_run_id` and moved the initialization to `createWorkflowContext`